hubspot3/test/test_contacts.py

```diff
@@ -1,112 +1,319 @@
+        mock_connection.assert_has_request(
+            "POST", "/contacts/v1/contact/email/{}/profile?".format(email), data
+        )
+        assert resp == response_body
+
+    def test_merge(self, contacts_client, mock_connection):
+        primary_id, secondary_id = 10, 20
+        mock_connection.set_response(200, "SUCCESS")
+        resp = contacts_client.merge(primary_id, secondary_id)
+        mock_connection.assert_num_requests(1)
+        mock_connection.assert_has_request(
+            "POST",
+            "/contacts/v1/contact/merge-vids/{}/?".format(primary_id),
+            dict(vidToMerge=secondary_id),
+        )
+        assert resp is None
+
+    @pytest.mark.parametrize(
+        "limit, query_limit, extra_properties",
+        [
+            (0, 100, None),
+            (10, 10, None),
+            (20, 20, None),
+            (150, 100, None),  # keep query limit at max 100
+            (10, 10, ["hs_analytics_last_url", "hs_analytics_revenue"]),
+            (10, 10, "lead_source"),
+        ],
+    )
+    def test_get_all(
+        self, contacts_client, mock_connection, limit, query_limit, extra_properties
+    ):
+        response_body = {
+            "contacts": [
+                {
+                    "addedAt": 1390574181854,
+                    "vid": 204727,
+                    "canonical-vid": 204727,
+                    "merged-vids": [],
+                    "properties": {},
+                }
+            ],
+            "has-more": False,
+            "vid-offset": 204727,
+        }
+        get_batch_return = [dict(id=204727 + i) for i in range(30)]
+        get_batch_mock = Mock(return_value=get_batch_return)
+        contacts_client.get_batch = get_batch_mock
+        mock_connection.set_response(200, response_body)
+        resp = contacts_client.get_all(limit=limit, extra_properties=extra_properties)
+        mock_connection.assert_num_requests(1)
+        mock_connection.assert_has_request(
+            "GET",
+            "/contacts/v1/lists/all/contacts/all?count={}&vidOffset=0".format(
+                query_limit
+            ),
+        )
+        assert resp == get_batch_return if not limit else get_batch_return[:limit]
+        get_batch_mock.assert_called_once_with(
+            [contact["vid"] for contact in response_body["contacts"]],
+            extra_properties=extra_properties,
+        )
+
+    @pytest.mark.parametrize(
+        "extra_properties_given, extra_properties_as_list",
+        [
+            (None, []),
+            ("lead_source", ["lead_source"]),
+            (
+                ["hs_analytics_last_url", "hs_analytics_revenue"],
+                ["hs_analytics_last_url", "hs_analytics_revenue"],
+            ),
+        ],
+    )
+    def test_get_batch(
+        self,
+        contacts_client,
+        mock_connection,
+        extra_properties_given,
+        extra_properties_as_list,
+    ):
+        response_body = {
```

Comment:
Du verlässt dich hier im Zusammenhang mit der letzten Zeile wieder einmal auf die Sortierung eines `dict`s, weswegen dieser Test ab und zu fehlschlägt. Zur Definition dieses Responses sollte daher besser ein `OrderedDict` verwendet werden.

Reply:
Hm, Kommando vllt. doch nochmal so halb zurück. Die Einführung des `OrderedDict`s hilft leider nur bis zu einem bestimmten Punkt: das Ganze wird dann ja serialisiert und in den Tiefen der `base.py` wieder deserialisiert. Beim Deserialisieren entsteht natürlich wieder ein normales `dict`, was natürlich wieder dasselbe Reihenfolge-Problem mit sich bringt.

Ich würde daher, um das möglichst simpel zu lösen, einfach vorschlagen, hier der Einfachheit halber wieder auf das normale `dict` umzustellen und stattdessen einfach das
```python
assert resp == [{"id": 3234574}, {"id": 3234575}]
```
am Ende anzupassen zu
```python
assert len(resp) == 2
assert {"id": 3234574} in resp
assert {"id": 3234575} in resp
```
Damit verlässt sich der Test gar nicht mehr auf irgendeine Reihenfolge.